packages/tools/devtools/devtools-view/src/DevtoolsView.tsx

```diff
@@ -170,22 +171,43 @@ export function DevtoolsView(): React.ReactElement {
 			handleIncomingMessage(message, inboundMessageHandlers, {
 				context: loggingContext,
 			});
+			clearTimeout(timeoutId); // Clear the timeout when cleaning up
 		}
 
 		messageRelay.on("message", messageHandler);
 
 		// Query for supported feature set
 		messageRelay.postMessage(getSupportedFeaturesMessage);
 
+		// Add a timeout for the query
+		const timeoutId = setTimeout(() => {
```

Comment:
Nit: For the variable name, this isn't really an ID, it's an actively running timer. Should we just call it "timer" (or something)?